### PR TITLE
Gesture data structures

### PR DESCRIPTION
**Pull request by rpavlik**
These were pulled out of the massive gesture pull request and subsequent modifications. They seem to be reasonable now, still need unit tests for the registered string map. @mars979 ?

### CODE REVIEW

**Comment by mars979** [comment]
Added the unit tests on registered string map and correlated map as well. Tests are passing

File: tests/cplusplus/Common/RegStringMap.cpp [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+#include <string>
+
+using osvr::util::StringID;
+using osvr::util::PeerStringID;
+using osvr::common::RegisteredStringMap;
+using osvr::common::CorrelatedStringMap;
+
+class RegisteredStringMapTest : public ::testing::Test {
+  public:
+    virtual void SetUp() {
```
Comment (rpavlik) [inline tests/cplusplus/Common/RegStringMap.cpp]: Can just use constructor, safer and easier.

File: tests/cplusplus/Common/RegStringMap.cpp [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+TEST(RegisteredStringMap, create) {
+
+    RegisteredStringMap regMap;
+    ASSERT_EQ(0, regMap.getEntries().size());
+    ASSERT_FALSE(regMap.isModified());
+}
+
+TEST(RegisteredStringMap, addNewValues) {
+    RegisteredStringMap regMap;
+    CorrelatedStringMap corMap;
```
Comment (rpavlik) [inline tests/cplusplus/Common/RegStringMap.cpp]: line not used?

File: tests/cplusplus/Common/RegStringMap.cpp [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    ASSERT_EQ(3, regMap.getEntries().size());
+    ASSERT_EQ(2, regID.value());
+}
+
+TEST_F(RegisteredStringMapTest, getValues) {
+
+    ASSERT_STREQ("RegVal0", regMap.getStringFromId(regID0).c_str());
+    ASSERT_STREQ("RegVal1", regMap.getStringFromId(regID1).c_str());
+    ASSERT_STREQ("RegVal2", regMap.getStringFromId(regID2).c_str());
+    ASSERT_EQ(0, std::strlen(regMap.getStringFromId(StringID(1000)).c_str()));
```
Comment (rpavlik) [inline tests/cplusplus/Common/RegStringMap.cpp]: can just `ASSERT_TRUE(regMap.getStringFromId(StringID(1000)).empty());`

File: tests/cplusplus/Common/RegStringMap.cpp [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    ASSERT_FALSE(regMap.isModified());
+    ASSERT_EQ(3, regMap.getEntries().size());
+    ASSERT_EQ(2, regID.value());
+}
+
+TEST_F(RegisteredStringMapTest, getValues) {
+
+    ASSERT_STREQ("RegVal0", regMap.getStringFromId(regID0).c_str());
+    ASSERT_STREQ("RegVal1", regMap.getStringFromId(regID1).c_str());
+    ASSERT_STREQ("RegVal2", regMap.getStringFromId(regID2).c_str());
```
Comment (rpavlik) [inline tests/cplusplus/Common/RegStringMap.cpp]: Should just be able to `ASSERT_EQ` without the `.c_str()`

**Comment by rpavlik** [comment]
Can merge once the few changes are done.

File: tests/cplusplus/Common/RegStringMap.cpp (hunk 1 above)
Comment (rpavlik) [inline tests/cplusplus/Common/RegStringMap.cpp]: Also, I'd move this fixture down to just before the tests that use it.

Comment (mars979) [inline tests/cplusplus/Common/RegStringMap.cpp]: Ok, using constructor now and moved fixture down before tests.

File: tests/cplusplus/Common/RegStringMap.cpp (hunk 2 above)
Comment (mars979) [inline tests/cplusplus/Common/RegStringMap.cpp]: Yup, copy paste err. Fixed now

File: tests/cplusplus/Common/RegStringMap.cpp (hunk 3 above)
Comment (mars979) [inline tests/cplusplus/Common/RegStringMap.cpp]: Fixed

File: tests/cplusplus/Common/RegStringMap.cpp (hunk 4 above)
Comment (mars979) [inline tests/cplusplus/Common/RegStringMap.cpp]: Yes, `ASSERT_STREQ` is only for c style strings. I didn't know that `ASSERT_EQ` would work for c++ strings, but it does so those are changed.

**Comment by mars979** [comment]
Modified the tests according to comments above, ready for merge now

**Pull request merged**